vm/src/memory/operand_stack.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <exception>
+#include <memory>
+
+class OperandStack {
+private:
+    char *bytes;
+    size_t capacity;
+    size_t current_pos;
+
+public:
+    explicit OperandStack(size_t len);
+
+    void push_int(int64_t i);
+
+    void push_double(double d);
+
+    void push_string_constant_ref(int64_t s);
+
+    int64_t pop_int();
+
+    double pop_double();
+
+    int64_t pop_string_constant_ref();
+
+    void pop_bytes(size_t size);
+
+    ~OperandStack();
+};
+
+struct OperandStackError : public std::exception {
```

Comment:
```suggestion
struct OperandStackError : public std::out_of_range {
```

Reply:
the `OperandStackError` is a kind of error for anything relied to the operand stack; its kind is not bound to any oor stuff

Reply:
This exception is only used when bound checking fails. If more error cases are added in the future, more exceptions types would just be created.